File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    pq = deque(priorities)
    lq = [i for i in range(len(priorities))]
    lq = deque(lq)
    #print(pq, lq)
    maximum = max(pq)
    while pq:
        pnum = pq.popleft()
        lnum = lq.popleft()
        if pnum == maximum:
            if lnum == location:
                answer += 1
                return answer                    
            else:
                answer += 1
                maximum = max(pq)
        else:
            pq.append(pnum)
            lq.append(lnum)
    
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    
    deque<int> pq(priorities.begin(), priorities.end());
    deque<int> lq;
    for (int i=0;i<priorities.size();i++){
        lq.push_back(i);
    }
    
    int pmax = *max_element(pq.begin(), pq.end());
    while (!pq.empty()){
        if (pq.front() == pmax){
            answer++;
            if (lq.front() == location){
                return answer;
            }
            else{
                pq.pop_front();
                lq.pop_front();
                pmax = *max_element(pq.begin(), pq.end());
            }
        }
        else{
            pq.push_back(pq.front());
            pq.pop_front();
            lq.push_back(lq.front());
            lq.pop_front();
        }
    }
    
    
    
    return answer;
}